src/openforms/forms/migrations/0072_check_form_variable_datatype.py

```diff
@@ -0,0 +1,36 @@
+# Generated by Django 3.2.18 on 2023-03-06 07:50
+
+from django.db import migrations
+
+from openforms.formio.utils import iter_components
+from openforms.variables.constants import FormVariableDataTypes
+
+
+def check_form_variable_datatype(apps, schema_editor):
+    FormVariable = apps.get_model("forms", "FormVariable")
+    form_variables = FormVariable.objects.filter(
+        data_type=FormVariableDataTypes.datetime
+    )
+
+    form_variables_to_update = []
+    for form_var in form_variables:
+        updated_form_var = False
+
+        form_def = form_var.form_definition
+        for comp in iter_components(configuration=form_def.configuration):
+            if comp["key"] == "date":
```

Comment:
this does not seem to correctly correlate form variables and the components from the form definition within the same form.

Try setting up test data like this:

1. form 1, has a variable with `key=field1`, `type=datetime`
2. form definition 1, used in form 1, has a component `{"type": "datetime", key: "field1"}`
3. form 2, has a variable with `key=field1`, `type=datetime`
4. form definition 2, used in form 2, has a component `{"type": "date", key: "field1"}`

After migrating, variable one should still be type `datetime`, while variable 2 should be type `date`

Reply:
This works okay because of the `FormVariable.form_definition` field, however, what will make it crash is setting up a user defined variable (non-component) with `type="datetime"`, since `form_def.configuration` will throw `AttributeError` (`form_def` will be None` in that case.